theano/gpuarray/c_code/dnn_fwd.c

```diff
@@ -142,6 +170,12 @@ APPLY_SPECIFIC(conv_fwd)(PyGpuArrayObject *input, PyGpuArrayObject *kerns,
           return 1;
         }
         algo = choice.algo;
+        prev_algo.algo = (int)algo;
+        prev_algo.wsSize = worksize = choice.memory;
+        prev_algo.mathType = mathtype = choice.mathType;
```

Comment:
@borisfom In fact, you use `choice.mathType` here, and `choice` is a `cudnnConvolutionFwdAlgoPerf_t`, but this structure has field `mathType` on cuDNN 7 only, not on cuDNN 6 and earlier.

So, when compiling with cuDNN < 7, this line generates an error.

On cuDNN 7:
```C
typedef struct {
    cudnnConvolutionFwdAlgo_t   algo;
    cudnnStatus_t               status;
    float                       time;
    size_t                      memory;
    cudnnDeterminism_t          determinism;
    cudnnMathType_t             mathType; // Field mathType is defined.
    int                         reserved[3];
} cudnnConvolutionFwdAlgoPerf_t;
```

On cuDNN 6:
```C
typedef struct {
    cudnnConvolutionFwdAlgo_t   algo;
    cudnnStatus_t               status;
    float                       time;
    size_t                      memory;
    cudnnDeterminism_t          determinism;
    // No mathType defined.
    int                         reserved[4];
} cudnnConvolutionFwdAlgoPerf_t;
```

Same for equivalent structures in `dnn_gw.c` and `dnn_gi.c`.